### PR TITLE
Edit the 'default_node_pool' and 'network_profile' config blocks

### PR DESCRIPTION
**Pull request by bamaralf**
Fixes #000 

Changes proposed in the pull request:
#63

### CODE REVIEW

**Review by yupwei68 — COMMENTED** [review]
Thanks for your contribution 🚀 ! A first look, please merge the latest changes from master branch, and not break our users by removing existing variables. Please add these enhanced fields into test/fixture and pass the tests. And add these enhanced fields into example in README.
After these changes, we'll have a better base to continue. 
Thanks!

File: README.md [hunk 1]
```diff
@@ -49,6 +48,10 @@ module "aks" {
   rbac_aad_managed                 = true
   private_cluster_enabled          = true # default value
 
+  network_profile                  = {
+    "network_plugin" = "kubenet"
+  }
+
```
[the remark this answers is not in the document]
Comment (bamaralf) [reply README.md]: Hi, I merged with the master branch

File: main.tf [hunk 2]
```diff
@@ -8,7 +8,8 @@ module "ssh-key" {
 }
 
 resource "azurerm_kubernetes_cluster" "main" {
-  name                    = "${var.prefix}-aks"
+  name                    = var.prefix
+  kubernetes_version      = var.kubernetes_version
```
[the remark this answers is not in the document]
Comment (bamaralf) [reply main.tf]: Hi, I merged with the master branch but I did not observe this 'kubernetes_version' PR. But the 'kubernetes_version 'config is in place.

File: variables.tf [hunk 3]
```diff
@@ -26,12 +26,6 @@ variable "admin_username" {
   type        = string
 }
 
-variable "agents_size" {
-  default     = "Standard_D2s_v3"
-  description = "The default virtual machine size for the Kubernetes agents"
-  type        = string
-}
-
```
[the remark this answers is not in the document]
Comment (bamaralf) [reply variables.tf]: I reversed these changes.

File: variables.tf [hunk 4]
```diff
@@ -68,18 +62,6 @@ variable "enable_log_analytics_workspace" {
...
-  default     = null
-}
-
-variable "os_disk_size_gb" {
-  description = "Disk size of nodes in GBs."
-  type        = number
-  default     = 50
-}
-
 variable "private_cluster_enabled" {
```
[the remark this answers is not in the document]
Comment (bamaralf) [reply variables.tf]: I reversed these changes.

File: variables.tf [hunk 5]
```diff
@@ -146,20 +141,87 @@ variable "rbac_aad_server_app_secret" {
...
   default     = null
 }
 
-variable "orchestrator_version" {
-  description = "Specify which Kubernetes release to use for the orchestration layer. The default used is the latest Kubernetes version available in the region"
-  type        = string
+variable "agents_min_count" {
+  type        = number
+  description = "Minimum number of nodes in a pool"
   default     = null
```
[the remark this answers is not in the document]
Comment (bamaralf) [reply variables.tf]: I merged with the master branch. This is in sync.

File: main.tf [hunk 6]
```diff
@@ -25,12 +26,22 @@ resource "azurerm_kubernetes_cluster" "main" {
...
-    os_disk_size_gb      = var.os_disk_size_gb
-    vnet_subnet_id       = var.vnet_subnet_id
+    orchestrator_version  = var.orchestrator_version == null ? var.kubernetes_version : var.orchestrator_version
+    name                  = var.agents_name
+    node_count            = var.agents_count
+    vm_size               = var.agents_vm_size
+    os_disk_size_gb       = var.agents_os_disk_size_gb
+    vnet_subnet_id        = var.agents_vnet_subnet_id
+    enable_node_public_ip = var.agents_enable_node_public_ip
+    enable_auto_scaling   = var.agents_enable_auto_scaling
```
[the remark this answers is not in the document]
Comment (bamaralf) [reply main.tf]: The PR is in place.

File: test/fixture/main.tf [hunk 7]
```diff
@@ -31,15 +31,13 @@ module "aks" {
...
-  orchestrator_version            = "1.19.3"
-  network_plugin                  = "azure"
   vnet_subnet_id                  = azurerm_subnet.test.id
-  os_disk_size_gb                 = 60
   enable_http_application_routing = true
   enable_azure_policy             = true
   sku_tier                        = "Paid"
   enable_kube_dashboard           = true
   private_cluster_enabled         = true
+  agents_os_disk_size_gb          = 60
```
[the remark this answers is not in the document]
Comment (bamaralf) [reply test/fixture/main.tf]: I updated the tests.

**Review by yupwei68 — COMMENTED** [review]
Thanks for your changes of most `ForceNew` fields enhancements. I have done some changes. Would you integrate the fields in example from test/fixture into example in README? After that, I think it will be good to me.

**Comment by bamaralf** [comment]
Hi @yupwei68. I updated the README.md

**Review by yupwei68 — APPROVED** [review]
LGTM!

**Pull request merged**